chapter09/axp/ex02

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env perl
+
+use 5.014;
+use warnings;
+
+my $fname = $ARGV[0] // 'patterns.dat';
+my $finish = $^O eq "MSWin32" ? "<Ctrl>+Z" : "<Ctrl>+D";
+
+open my $file, $fname or die "Couldn't read $fname: $!\n";
+my @pats = map { chomp; qr/$_/ } <$file>;
+close $file;
+
+say "Patterns";
+say "--------";
+say foreach @pats;
+say "\nNow type lines of text; $finish to finish.";
+
+while ( <> ) {
+	foreach my $pattern ( @pats ) {
+		print "$.\t${^PREMATCH}<<<${^MATCH}>>>${^POSTMATCH}" while ( /$pattern/g );
```

Comment:
These named match variables are useful, but don't use them in the DPG as they have an impact on performance.